sympy/combinatorics/galois.py

```diff
@@ -0,0 +1,589 @@
+
+    all_names = [
+        C6, S3, D6, A4, G18, A4xC2, S4m, S4p,
+        G36m, G36p, S4xC2, PSL2F5, G72, PGL2F5, A6, S6
+    ]
+
+
+def get_group_by_name(degree, name):
+    r"""
+    Get a :py:class:`~.PermutationGroup` by its name.
+
+    Parameters
+    ==========
+
+    degree : int
+        That $n$ such that the desired group is a subgroup of $S_n$.
+
+    name : string
+        The name of the group. Should be a value of one of the
+        ``SnTransitiveSubgroups`` enum classes, for ``n`` equal to *degree*.
+
+    Returns
+    =======
+
+    :py:class:`~.PermutationGroup`
+
+    """
+    if degree == 1:
+        return SymmetricGroup(1)
+
+    elif degree == 2:
+        return SymmetricGroup(2)
+
+    elif degree == 3:
+        cls = S3TransitiveSubgroups
+        if name == cls.A3:
+            return AlternatingGroup(3)
+        elif name == cls.S3:
+            return SymmetricGroup(3)
+
+    elif degree == 4:
+        cls = S4TransitiveSubgroups
+        if name == cls.C4:
+            return CyclicGroup(4)
+        elif name == cls.V:
+            return four_group()
+        elif name == cls.D4:
+            return DihedralGroup(4)
+        elif name == cls.A4:
+            return AlternatingGroup(4)
+        elif name == cls.S4:
+            return SymmetricGroup(4)
+
+    elif degree == 5:
+        cls = S5TransitiveSubgroups
+        if name == cls.C5:
+            return CyclicGroup(5)
+        elif name == cls.D5:
+            return DihedralGroup(5)
+        elif name == cls.M20:
+            return M20()
+        elif name == cls.A5:
+            return AlternatingGroup(5)
+        elif name == cls.S5:
+            return SymmetricGroup(5)
+
+    elif degree == 6:
+        cls = S6TransitiveSubgroups
+        if name == cls.C6:
+            return CyclicGroup(6)
+        elif name == cls.S3:
+            return S3_in_S6()
+        elif name == cls.D6:
+            return DihedralGroup(6)
+        elif name == cls.A4:
+            return A4_in_S6()
+        elif name == cls.G18:
+            return G18()
+        elif name == cls.A4xC2:
+            return A4xC2()
+        elif name == cls.S4m:
+            return S4m()
+        elif name == cls.S4p:
+            return S4p()
+        elif name == cls.G36m:
+            return G36m()
+        elif name == cls.G36p:
+            return G36p()
+        elif name == cls.S4xC2:
+            return S4xC2()
+        elif name == cls.PSL2F5:
+            return PSL2F5()
+        elif name == cls.G72:
+            return G72()
+        elif name == cls.PGL2F5:
+            return PGL2F5()
+        elif name == cls.A6:
+            return AlternatingGroup(6)
+        elif name == cls.S6:
+            return SymmetricGroup(6)
+
+
+def four_group():
+    """
+    Return a representation of the Klein four-group as a transitive subgroup
+    of S4.
+    """
+    return PermutationGroup(
+        Permutation(0, 1)(2, 3),
+        Permutation(0, 2)(1, 3)
+    )
+
+
+def M20():
```

Comment:
I think that `snake_case` naming should be encouraged.

Reply:
I would make an exception when it names a mathematical value written with a specific convention, like this presumably is.

Reply:
Also an important coding convention to use in SymPy is that when an algorithm is written from a book or paper, as is the case here, one should always try to match the algorithm to the paper with things like variable names and so on. That makes it much easier to cross-reference the book or paper in the future if there is ever some issue.

Reply:
it is less important for denotational problem because snakecase-camelcase transformation is just capitalizing the words over it

Reply:
And I assume that hyphens in things like `S3_in_S6` are just a workaround for spacing than to match exactly the original denotation. So that argument neither wouldn't strictly follow.

Reply:
I do feel that mathematical conventions are more important than Python conventions in cases like these (although I understand the desire to have a uniform codebase).

For a different but related example, PEP8 recommends that variables always be lowercase, but that would be unthinkable in SymPy. (How could, say, a group and its element not be `G` and `g`?)

Reply:
I think that capital with hyphens could closely fit to `CONSTANT_CASE` (for example, `S3_IN_S6`) 
if you are happy with it.